### PR TITLE
[Uploader] Add nexus uploader

### PR DESCRIPTION
**Pull request by Herklos**
Depends on #87

### CODE REVIEW

File: octobot_tentacles_manager/uploaders/nexus_uploader.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            if destination_folder_name is not None else os.path.basename(folder_path)
+        upload_folder_url: str = f"{self.nexus_url}/{upload_path}/{dest_folder_name}"
+        for file_path in os.listdir(folder_path):
+            upload_file_url = f"{upload_folder_url}/{file_path}"
+            self.logger.debug(f"Uploading {file_path} to nexus at {upload_file_url}...")
+            error_count += await self._upload(file_url_on_nexus=upload_file_url,
+                                              local_file_path=os.path.join(folder_path, file_path))
+        return error_count
+
+    async def _upload(self, file_url_on_nexus: str, local_file_path: str):
```
Comment (GuillaumeDSM) [inline octobot_tentacles_manager/uploaders/nexus_uploader.py:67]: 💯

**Review by GuillaumeDSM — APPROVED** [review]
great !

File: octobot_tentacles_manager/api/creator.py [hunk 2]
```diff
@@ -76,7 +81,24 @@ async def create_all_tentacles_bundle(output_dir: str = constants.DEFAULT_EXPORT
...
                 should_remove_artifacts_after_use=should_remove_artifacts_after_use)
             await tentacle_bundle_exporter.export()
+            tentacle_bundle_exported_list.append(tentacle_bundle_exporter)
         except Exception as e:
             logger.error(f"Error when exporting tentacle {tentacle.name} : {str(e)}")
             error_count += 1
+
+    if upload_url is not None:
+        for exported_tentacle_bundle in tentacle_bundle_exported_list:
+            await _upload_exported_tentacle_bundle(upload_url, exported_tentacle_bundle)
```
Comment (GuillaumeDSM) [inline octobot_tentacles_manager/api/creator.py:91]: should we use 
`await asyncio.gather(*[
    _upload_exported_tentacle_bundle(upload_url, exported_tentacle_bundle)
    for exported_tentacle_bundle in tentacle_bundle_exported_list
])`
to use the uploaded asynchronousness ?

Comment (Herklos) [reply octobot_tentacles_manager/api/creator.py:91]: Good idea :+1:

File: tests/uploaders/test_nexus_uploader.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    # test upload file
+    with open(local_file_name, "w") as test_file:
+        test_file.write(json.dumps("{'test-key': 1}"))
+    assert await uploader_api.upload_file_or_folder_to_nexus(nexus_path=TEST_NEXUS_PATH,
+                                                             artifact_path=local_file_name,
+                                                             artifact_alias=nexus_test_file_name) == 0
+    # test download file
+    downloaded_file_path: str = await download_file_from_nexus(f"{TEST_NEXUS_PATH}/{nexus_test_file_name}",
+                                                               "downloaded_file")
```
Comment (GuillaumeDSM) [inline tests/uploaders/test_nexus_uploader.py:60]: 👍

File: tests/uploaders/test_nexus_uploader.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    # with open(downloaded_file_path, "r") as downloaded_file:
+    #     assert json.loads(downloaded_file.read()) == {
+    #         'test-key': 1
+    #     }
```
Comment (GuillaumeDSM) [inline tests/uploaders/test_nexus_uploader.py:65]: wip ?

Comment (Herklos) [reply tests/uploaders/test_nexus_uploader.py:65]: Yes

**Pull request merged**